src/main/java/net/snowflake/client/core/SFSessionProperty.java

```diff
@@ -56,7 +56,8 @@ public enum SFSessionProperty {
   PRIVATE_KEY_FILE("private_key_file", false, String.class),
   PRIVATE_KEY_FILE_PWD("private_key_file_pwd", false, String.class),
   CLIENT_INFO("snowflakeClientInfo", false, String.class),
-  ALLOW_UNDERSCORES_IN_HOST("allowUnderscoresInHost", false, Boolean.class);
+  ALLOW_UNDERSCORES_IN_HOST("allowUnderscoresInHost", false, Boolean.class),
+  USER_AGENT_IDENTIFIER("user_agent_identifier", false, String.class);
```

Comment:
I wouldn't call it "identifier" since either the user is already considering the "user-agent" itself to be an identifier, or else they'll interpret this to mean something like "identifying a user agent".

What we really are conveying is the addition of a user-agent suffix that can be used by the caller for all the "usual" purposes of user-agents, which includes but is not limited to identifying a client application.

Maybe "user_agent_suffix" would be most in-line. Examples:

https://stackoverflow.com/questions/57828760/how-to-set-user-agent-prefix-for-every-upload-request-to-s3-from-amazon-emr-ap
`fs.s3a.user.agent.prefix` for the open-source S3 Hadoop FileSystem
Putting `userAgentTemplate` in `awssdk_config_default.json`
`ClientConfiguration.withUserAgentPrefix` in AWS SDK
`fs.gs.application.name.suffix` for the open-source GCS Hadoop FileSystem: https://docs.cloudera.com/HDPDocuments/HDP2/HDP-2.6.5/bk_cloud-data-access/content/gcp-suffix.html

Reply:
Done